packages/plugin-session-replay-browser/src/session-replay.ts

```diff
@@ -108,6 +142,20 @@ class SessionReplay implements SessionReplayEnrichmentPlugin {
     }
   }
 
+  setShouldRecord(sessionStore?: IDBStoreSession) {
+    if (sessionStore?.shouldRecord === false) {
+      this.shouldRecord = false;
+    } else if (this.options && this.options.sampleRate) {
+      this.shouldRecord = Math.random() < this.options.sampleRate;
```

Comment:
Will this result essentially in the _next_ session being selected or unselected for recording?

Reply:
No, it'll capture that the current session should not be recorded, like this:
1. On plugin initialization, if user has a matching session in the indexdb store, we'll check there for if shouldRecord should be set to false
2. If they don't have a matching session in the store (even if they are in the middle of a session, ie if a customer has just set up the plugin on their site), we'll calculate it based on the sample rate
3. Otherwise we'll check if optOut is configured

We'll do all of the same things when session_start fires, telling us that a new session has begun and we should recalculate whether to record or not.

Reply:
I wonder if config.optOut should override any other values actually, I think it should, so I'll move it to be the first check in this if statement

Reply:
it's starting to feel like we have more states on `this` that we need to manage/cleanup when a new session starts.